src/QoiSharp/QoiImage.cs

```diff
@@ -10,8 +10,12 @@ public class QoiImage
     /// <summary>
     /// Raw pixel data.
     /// </summary>
-    public byte[] Data { get; }
-    
+    public ReadOnlyMemory<byte> Data { get; }
```

Comment:
This is an API breaking change. I'd love it if you'd break API and make this change as public mutable APIs are a problematic. This makes it clear that the buffer is intended to be readonly.